Layers/Renderers/Building/BuildingRenderer.gd

```diff
@@ -94,28 +136,158 @@ func load_feature_instance(feature):
 		color.v *= 0.9
 		color.s *= 1.6
 
-		roof.set_color(color)
+		roof.color = color
 
 		building.add_child(roof)
 
 	# Set parameters in the building base
-	building.set_footprint(polygon)
-	building.set_holes(holes)
-	building.set_offset(center[0], center[1])
-
+	building.set_metadata(building_metadata)
+	building.position = building_metadata["engine_center_position"]
 	building.name = str(feature.get_id())
 
 	# Build!
 	building.build()
 
-	building.position.y = layer_composition.render_info.ground_height_layer.get_value_at_position(
-		building.get_center().x + center[0],
-		-building.get_center().z + center[1]
-	) - cellar_height
-
 	return building
 
 
+func prepare_pillars(building_metadata: Dictionary, building: Node3D, num_floors: int):
+	var walls_scene = load("res://Buildings/Components/Walls/Pillars.tscn").instantiate()
+	walls_scene.ground_height_at_center = building_metadata["engine_center_position"].y
+	walls_scene.floors = num_floors
+	building.add_child(walls_scene)
+
+
+func prepare_plain_walls(building_type: String, building_metadata: Dictionary,
+		building: Node3D, num_floors: int):
+	var walls_scene = preload("res://Buildings/Components/Walls/PlainWalls.tscn")
+	var walls_resource: PlainWallResource = wall_resources[int(building_type)] \
+		if int(building_type) in range(0, wall_resources.size()) \
+		else fallback_wall
+	
+	# Random facade texture
+	var random_gen = RandomNumberGenerator.new()
+	random_gen.seed = hash(building_metadata["footprint"])
+
+	var wall_color = Color.WHITE_SMOKE
+	var random = random_gen.randi_range(0, 10)
```

Comment:
would be a good opportunity to make this more generic as well by making this work with any number of random colors rather than a fixed amount and probabilities. could add `random_color_weights` or something like that

Reply:
(not a blocker for merging though, can be done later)